src/Utils/SchemaExtender.php

```diff
@@ -0,0 +1,627 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Utils;
+
+use GraphQL\Error\Error;
+use GraphQL\Language\AST\DirectiveDefinitionNode;
+use GraphQL\Language\AST\DocumentNode;
+use GraphQL\Language\AST\Node;
+use GraphQL\Language\AST\NodeKind;
+use GraphQL\Language\AST\ObjectTypeExtensionNode;
+use GraphQL\Language\AST\SchemaDefinitionNode;
+use GraphQL\Language\AST\SchemaTypeExtensionNode;
+use GraphQL\Language\AST\TypeDefinitionNode;
+use GraphQL\Language\AST\TypeExtensionNode;
+use GraphQL\Type\Definition\CustomScalarType;
+use GraphQL\Type\Definition\Directive;
+use GraphQL\Type\Definition\EnumType;
+use GraphQL\Type\Definition\EnumValueDefinition;
+use GraphQL\Type\Definition\FieldArgument;
+use GraphQL\Type\Definition\InputObjectType;
+use GraphQL\Type\Definition\InterfaceType;
+use GraphQL\Type\Definition\ListOfType;
+use GraphQL\Type\Definition\NamedType;
+use GraphQL\Type\Definition\NonNull;
+use GraphQL\Type\Definition\ObjectType;
+use GraphQL\Type\Definition\Type;
+use GraphQL\Type\Definition\UnionType;
+use GraphQL\Type\Introspection;
+use GraphQL\Type\Schema;
+use GraphQL\Validator\DocumentValidator;
+use function array_keys;
+use function array_map;
+use function array_merge;
+use function array_values;
+use function count;
+use function is_array;
+
+class SchemaExtender
+{
+    const SCHEMA_EXTENSION = 'SchemaExtension';
+
+    /** @var Type[] */
+    protected static $extendTypeCache;
+
+    /** @var mixed[] */
+    protected static $typeExtensionsMap;
+
+    /** @var ASTDefinitionBuilder */
+    protected static $astBuilder;
+
+    /**
+     * @return TypeExtensionNode[]|null
+     */
+    protected static function getExtensionASTNodes(NamedType $type) : ?array
+    {
+        if (! isset($type->name)) {
```

Comment:
This line generates the error
```
Accessing name on the interface GraphQL\Type\Definition\NamedType suggest that you code against a concrete implementation. How about adding an instanceof check?
```
How should I handle this? There seems to be no interface that provides the "name" property